include/swift/AST/DiagnosticsSema.def

```diff
@@ -2931,8 +2931,8 @@ ERROR(dynamic_self_non_method,none,
       "%select{global|local}0 function cannot return 'Self'", (bool))
 
 ERROR(dynamic_self_invalid,none,
-      "covariant 'Self' can only appear as the type of a property, subscript or method result; "
-      "did you mean '%0'?", (StringRef))
+      "covariant 'Self' can only appear as the possibly optional type of a "
```

Comment:
Does this sentence mean simply covariant `Self` or `Self?`? I struggle a little with how to parse “possibly optional type.”

Reply:
The intention was to convey that both `Self` or `Self?` are supported. Do you happen to have a suggestion for a better wording?

Reply:
Perhaps simply: "covariant 'Self' or 'Self?' can only appear as..."

Reply:
Heh, indeed. Thank you, I'll take care of this!